src/integration-test/integration/ReportsClient.test.ts

```diff
@@ -0,0 +1,229 @@
+    // create mappings for tests
+    const newMapping: MappingCreate = {
+      mappingName: "Test1",
+    };
+    let mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    expect(mapping.mappingName).to.be.eq("Test1");
+    mappingIds.push(mapping.id);
+
+    newMapping.mappingName = "Test2";
+    mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    expect(mapping.mappingName).to.be.eq("Test2");
+    mappingIds.push(mapping.id);
+
+    newMapping.mappingName = "Test3";
+    mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    expect(mapping.mappingName).to.be.eq("Test3");
+    mappingIds.push(mapping.id);
+
+    // create reports for tests
+    const newReport: ReportCreate = {
+      displayName: "Test1",
+      projectId,
+    };
+    let report = await reportsClient.createReport(accessToken, newReport);
+    expect(report).to.not.be.undefined;
+    expect(report.displayName).to.be.eq("Test1");
+    reportIds.push(report.id);
+
+    newReport.displayName = "Test2";
+    report = await reportsClient.createReport(accessToken, newReport);
+    expect(report).to.not.be.undefined;
+    expect(report.displayName).to.be.eq("Test2");
+    reportIds.push(report.id);
+
+    newReport.displayName = "Test3";
+    report = await reportsClient.createReport(accessToken, newReport);
+    expect(report).to.not.be.undefined;
+    expect(report.displayName).to.be.eq("Test3");
+    reportIds.push(report.id);
+
+    // create reportMappings for tests
+    const newReportMapping: ReportMappingCreate = {
+      mappingId: mappingIds[mappingIds.length-3],
+      imodelId: testIModel.id,
+    };
+    let reportMapping = await reportsClient.createReportMapping(accessToken, reportIds[reportIds.length-1], newReportMapping);
+    expect(reportMapping).to.not.be.undefined;
+    expect(reportMapping.mappingId).to.be.eq(mappingIds[mappingIds.length-3]);
+    reportMappingIds.push(reportMapping.mappingId);
+
+    newReportMapping.mappingId = mappingIds[mappingIds.length-2];
+    reportMapping = await reportsClient.createReportMapping(accessToken, reportIds[reportIds.length-1], newReportMapping);
+    expect(reportMapping).to.not.be.undefined;
+    expect(reportMapping.mappingId).to.be.eq(mappingIds[mappingIds.length-2]);
+    reportMappingIds.push(reportMapping.mappingId);
+
+    newReportMapping.mappingId = mappingIds[mappingIds.length-1];
+    reportMapping = await reportsClient.createReportMapping(accessToken, reportIds[reportIds.length-1], newReportMapping);
+    expect(reportMapping).to.not.be.undefined;
+    expect(reportMapping.mappingId).to.be.eq(mappingIds[mappingIds.length-1]);
+    reportMappingIds.push(reportMapping.mappingId);
+  });
+
+  after(async function () {
+    while(reportMappingIds.length > 0) {
```

Comment:
Deleting a report should be enough. You do not need to delete report mappings before you delete the report.

Reply:
Removed the deletion of report mappings